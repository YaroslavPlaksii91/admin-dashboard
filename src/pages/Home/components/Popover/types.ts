import { ReactNode } from 'react';

export type PopoverProps = {
  id: string | undefined;
  open: boolean;
  anchorEl: HTMLButtonElement | null;
  handleClose: () => void;
  children: ReactNode;
};
