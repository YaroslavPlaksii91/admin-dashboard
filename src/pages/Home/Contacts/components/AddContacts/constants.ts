import { IMAGE_SIZE } from '@components/AddPhoto/constants';

export const ADD_CONTACT_FIELDS = Object.freeze({
  PHOTO: 'photo',
  FIRST_NAME: 'firstName',
  LAST_NAME: 'lastName',
  EMAIL: 'email',
  ADDRESS: 'address',
});

export const ADD_CONTACT_CONFIG = Object.freeze({
  [ADD_CONTACT_FIELDS.PHOTO]: {
    label: 'Add photo',
    name: 'photo',
    size: IMAGE_SIZE.BIG,
  },
  [ADD_CONTACT_FIELDS.FIRST_NAME]: {
    label: 'first name',
    type: 'text',
    name: 'firstName',
    placeholder: 'First name',
  },
  [ADD_CONTACT_FIELDS.LAST_NAME]: {
    label: 'last name',
    type: 'text',
    name: 'lastName',
    placeholder: 'Last name',
  },
  [ADD_CONTACT_FIELDS.EMAIL]: {
    label: 'email',
    type: 'email',
    name: 'email',
    placeholder: 'Email',
  },
  [ADD_CONTACT_FIELDS.ADDRESS]: {
    label: 'address',
    type: 'text',
    name: 'address',
    placeholder: 'Address',
  },
});
