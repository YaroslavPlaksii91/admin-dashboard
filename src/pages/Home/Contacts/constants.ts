export const CONTACTS_COLUMNS = Object.freeze([
  {
    title: 'Name',
    name: 'name',
    size: 4,
  },
  {
    title: 'Email',
    name: 'email',
    size: 3,
  },
  {
    title: 'Address',
    name: 'address',
    size: 3,
  },
  {
    title: 'Created at',
    name: 'created',
    size: 2,
  },
]);
