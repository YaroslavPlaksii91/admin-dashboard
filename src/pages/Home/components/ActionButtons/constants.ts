export const SORT_BTN_TEST_ID = 'sortButton';
export const FILTER_BTN_TEST_ID = 'filterButton';
export const ADD_BTN_TEST_ID = 'addButton';
