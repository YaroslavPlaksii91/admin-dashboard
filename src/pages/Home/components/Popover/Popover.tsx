import { FC } from 'react';
import { Popover as PopoverComponent } from '@mui/material';

import { PopoverProps } from './types';

export const Popover: FC<PopoverProps> = ({
  id,
  open,
  anchorEl,
  handleClose,
  children,
}) => {
  return (
    <PopoverComponent
      id={id}
      open={open}
      anchorEl={anchorEl}
      onClose={handleClose}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'left',
      }}
    >
      {children}
    </PopoverComponent>
  );
};
