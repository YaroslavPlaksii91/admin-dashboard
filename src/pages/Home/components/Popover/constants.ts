export const POPOVER_TEST_ID = 'popover';
