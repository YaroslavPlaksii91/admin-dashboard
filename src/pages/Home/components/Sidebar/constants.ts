export const USERS_PAGES = Object.freeze([
  {
    name: 'Overview',
    icon: '/src/assets/icons/sprite.svg#icon-overview',
    link: '/overview',
  },
  {
    name: 'Tickets',
    icon: '/src/assets/icons/sprite.svg#icon-tickets',
    link: '/tickets',
  },
  {
    name: 'Ideas',
    icon: '/src/assets/icons/sprite.svg#icon-ideas',
    link: '/ideas',
  },
  {
    name: 'Contacts',
    icon: '/src/assets/icons/sprite.svg#icon-contacts',
    link: '/contacts',
  },
  {
    name: 'Agents',
    icon: '/src/assets/icons/sprite.svg#icon-agents',
    link: '/agents',
  },
  {
    name: 'Articles',
    icon: '/src/assets/icons/sprite.svg#icon-articles',
    link: '/articles',
  },
]);

export const USERS_SETTINGS = Object.freeze([
  {
    name: 'Settings',
    icon: '/src/assets/icons/sprite.svg#icon-settings',
    link: '/settings',
  },
  {
    name: 'Subscription',
    icon: '/src/assets/icons/sprite.svg#icon-subscription',
    link: '/subscription',
  },
]);

export const SIDEBAR_LINK_TEST_ID = 'sidebarLink';
