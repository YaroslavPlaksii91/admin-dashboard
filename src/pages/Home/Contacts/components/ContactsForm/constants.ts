import { IMAGE_SIZE } from '@components/AddPhoto/constants';

export const CONTACTS_FORM_FIELDS = Object.freeze({
  PHOTO: 'photo',
  FIRST_NAME: 'firstName',
  LAST_NAME: 'lastName',
  EMAIL: 'email',
  ADDRESS: 'address',
});

export const CONTACTS_FORM_CONFIG = Object.freeze({
  [CONTACTS_FORM_FIELDS.PHOTO]: {
    label: 'Add photo',
    name: 'photo',
    size: IMAGE_SIZE.BIG,
  },
  [CONTACTS_FORM_FIELDS.FIRST_NAME]: {
    label: 'first name',
    type: 'text',
    name: 'firstName',
    placeholder: 'First name',
  },
  [CONTACTS_FORM_FIELDS.LAST_NAME]: {
    label: 'last name',
    type: 'text',
    name: 'lastName',
    placeholder: 'Last name',
  },
  [CONTACTS_FORM_FIELDS.EMAIL]: {
    label: 'email',
    type: 'email',
    name: 'email',
    placeholder: 'Email',
  },
  [CONTACTS_FORM_FIELDS.ADDRESS]: {
    label: 'address',
    type: 'text',
    name: 'address',
    placeholder: 'Address',
  },
});
