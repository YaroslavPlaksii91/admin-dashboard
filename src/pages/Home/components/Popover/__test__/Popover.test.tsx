import { cleanup, render, screen } from '@testing-library/react';

import { Popover } from '../Popover';
import { POPOVER_TEST_ID } from '../constants';

describe('Popover component', () => {
  afterEach(cleanup);

  const id = 'popover';
  const open = true;
  const anchorEl = null;
  const handleClose = jest.fn();

  it('makes snapshot', () => {
    const { asFragment } = render(
      <Popover
        id={id}
        open={open}
        anchorEl={anchorEl}
        handleClose={handleClose}
      >
        <div>Popover Content</div>
      </Popover>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be rendered correctly', () => {
    render(
      <Popover
        id={id}
        open={open}
        anchorEl={anchorEl}
        handleClose={handleClose}
      >
        <div>Popover Content</div>
      </Popover>,
    );

    const popover = screen.getByTestId(POPOVER_TEST_ID);
    const content = screen.getByText('Popover Content');

    expect(popover).toBeInTheDocument();
    expect(content).toBeInTheDocument();
  });
});
