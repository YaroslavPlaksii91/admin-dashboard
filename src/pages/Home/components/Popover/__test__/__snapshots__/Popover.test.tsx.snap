// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component makes snapshot 1`] = `<DocumentFragment />`;
